boards/nxp/imx95_evk/CMakeLists.txt

```diff
@@ -11,3 +11,47 @@ if (CONFIG_SOF AND CONFIG_BOARD_IMX95_EVK_MIMX9596_M7_DDR)
     DEPENDS ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME}
   )
 endif()
+
+if(CONFIG_BOARD_NXP_SPSDK_IMAGE OR (DEFINED ENV{CONFIG_BOARD_NXP_SPSDK_IMAGE}
+  AND "$ENV{CONFIG_BOARD_NXP_SPSDK_IMAGE}" STREQUAL "y"))
+  # The blobs binary is self-extracting file which could be used only in Linux bash
+  if(NOT CMAKE_HOST_UNIX)
+    message(FATAL_ERROR "SPSDK Image generating failed: requires Linux host!")
+  endif()
+  if(CONFIG_BOARD_IMX95_EVK_MIMX9596_M7)
+    set(FIRMWARE_RELEASE "imx95-19x19-lpddr5-evk-boot-firmware-0.1")
+    set(AHAB_CONFIG_FILE "imx95_evk_mimx9596_m7_ahab.yaml")
+  else()
+    message(FATAL_ERROR "SPSDK Image not supported on the platform!")
+  endif()
+  file(WRITE ${CMAKE_BINARY_DIR}/zephyr/${AHAB_CONFIG_FILE}
+    "
+    family: mimx9596
+    revision: a1
+    target_memory: standard
+    output:                   ${CMAKE_BINARY_DIR}/zephyr/flash.bin
+    containers:
+      - binary_container:
+          path:               ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/mx95a0-ahab-container.img
+      - container:
+          srk_set: none
+          images:
+            - lpddr_imem:     ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/lpddr5_imem_v202311.bin
+              lpddr_imem_qb:  ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/lpddr5_imem_qb_v202311.bin
+              lpddr_dmem:     ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/lpddr5_dmem_v202311.bin
+              lpddr_dmem_qb:  ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/lpddr5_dmem_qb_v202311.bin
+              oei_ddr:        ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/oei-m33-ddr.bin
+            - oei_tcm:        ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/oei-m33-tcm.bin
+            - system_manager: ${CMAKE_BINARY_DIR}/zephyr/${FIRMWARE_RELEASE}/m33_image-mx95alt.bin
+            - cortex_m7_app:  ${CMAKE_BINARY_DIR}/zephyr/zephyr.bin
+            - v2x_dummy: true
+    ")
+  set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+    COMMAND chmod +x ${ZEPHYR_HAL_NXP_MODULE_DIR}/zephyr/blobs/imx-boot-firmware/${FIRMWARE_RELEASE}.bin
```

Comment:
and actually it should be added to the repo with this already, do not chmod here

Reply:
The problem is the firmware which is a self-extractable archive is not from any repo. But from web site.
https://www.nxp.com.cn/lgfiles/NMG/MAD/YOCTO/imx95-19x19-lpddr5-evk-boot-firmware-0.1.bin

So, actually I have no idea. Not sure if there is other way to process it.

Reply:
@pdgendt @carlescufi @nashif what are your thoughts on this being a linux-only executable that needs a chmod?

Reply:
I'd be hesitant of running an executable downloaded during the build process. Would it be possible to have a different format for the required files?

I also opened https://github.com/zephyrproject-rtos/zephyr/issues/62250 a long time ago.

Reply:
Currently we still don't support the other archive format, so I think we can merge this PR firstly and provide update if a better archive format is available for us, thanks.